src/main/java/org/folio/innreach/batch/contribution/InitialContributionJobConsumerContainer.java

```diff
@@ -0,0 +1,127 @@
+package org.folio.innreach.batch.contribution;
+
+import feign.FeignException;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.folio.innreach.batch.contribution.listener.ContributionExceptionListener;
+import org.folio.innreach.batch.contribution.service.ContributionJobRunner;
+import org.folio.innreach.domain.dto.folio.inventorystorage.InstanceIterationEvent;
+import org.folio.innreach.external.exception.InnReachConnectionException;
+import org.folio.innreach.external.exception.ServiceSuspendedException;
+import org.folio.innreach.external.exception.SocketTimeOutExceptionWrapper;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.DefaultErrorHandler;
+import org.springframework.util.backoff.BackOff;
+import org.springframework.util.backoff.FixedBackOff;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.folio.innreach.batch.contribution.ContributionJobContextManager.getContributionJobContext;
+
+@Log4j2
+@Setter
+@RequiredArgsConstructor
+public class InitialContributionJobConsumerContainer {
+
+  public static final long POLL_TIMEOUT = 9000000L;
```

Comment:
This is 2.5 hours. It seems quite a bit higher than it needs to be. My understanding is:
* This value [needs to be greater](https://stackoverflow.com/questions/66108816/spring-kafka-2-6-5-infinite-retry-policy-via-stateful-retry-and-seektocurrenterr) than what we're setting as the `kafka.backoff.interval` (used as `retryConfig.getInterval`) but 2.5 hours is a big number.
* [The default of this value is 5 minutes](https://docs.spring.io/spring-kafka/api/org/springframework/kafka/listener/ConsumerProperties.html#setPollTimeout(long)).

@Arghya22 it seems like you think that 5 minutes isn't enough (if it is 5 minutes, the default). Was something else overriding the 5 minute default perhaps?

Is there a downside to keeping it at 2.5 hours? I guess that it could hide a broken Kafka retry process, keeping the container alive when we in fact don't want it to be stuck retrying, but I'm not sure.

Reply:
default time 5 min is enough and this big number will create no effect as now kafka itself is handling the retry with a interval of 90 sec . As a safety measure I added it , and surely it wont create any issue as its a max value.